### PR TITLE
Move some gnome libs to toplevel

### PR DESCRIPTION
**Pull request by worldofpeace**
###### Motivation for this change
cc @jtojnar 

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/libraries/folks/default.nix [hunk 1]
```diff
@@ -33,19 +32,14 @@ in stdenv.mkDerivation rec {
 
   passthru = {
     updateScript = gnome3.updateScript {
-      packageName = "folks";
-      attrPath = "gnome3.folks";
-      versionPolicy = "none";
+      packageName = pname;
```
Comment (jtojnar) [inline pkgs/development/libraries/folks/default.nix]: We should keep the version policy (we are using what seems to be unstable versions according to GNOME versioning).

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -4834,7 +4834,7 @@ in
   };
 
   pinentry_gnome = res.pinentry.override {
-    gcr = gnome3.gcr;
+    gcr = gcr;
```
Comment (jtojnar) [inline pkgs/top-level/all-packages.nix]: Can be changed to `inherit gcr`

File: pkgs/applications/misc/pmenu/default.nix [hunk 3]
```diff
@@ -13,7 +13,7 @@ stdenv.mkDerivation rec {
 
   nativeBuildInputs = [ python2Packages.wrapPython ];
 
-  buildInputs = [ python2Packages.pygtk gnome3.gnome-menus ];
+  buildInputs = [ python2Packages.pygtk ];
```
Comment (jtojnar) [inline pkgs/applications/misc/pmenu/default.nix]: `gnome-menus`

**Comment by worldofpeace** [comment]
Ok this eval fail is now from `hackage-packages.nix`. 
I'm not familiar with haskell in nixpkgs but I know that this file is autogenerated.

**Comment by jtojnar** [comment]
It is probably a good idea to inherit the libraries inside `gnome3` attrset, not only for the haskell but also so we could still update them with `nix-shell maintainers/scripts/update.nix --argstr path gnome3` (before we implement package tags).

**Comment by worldofpeace** [comment]
Always happy to help :+1:

**Pull request merged**